handler/README.md

```diff
@@ -1,232 +1,378 @@
+
+#### Graph
+Task logic is defined in the form of a graph. Actually, it is a tree structure. Each graph node sets 
+fragment logic to perform and outgoing edges, called Transitions. Its configuration looks like:
+```hocon
+graph {
+  # node logic options
+  node { }
+  # node outgoing edges
+  onTransitions {
+    _success { }
+    _error { }
+    someCustomTransition { }
+  }
+}
+```
+There are two sections:
+- `node`defines a fragment processing logic
+- `onTransitions` is a map that represents outgoing edges in a graph
+
+#### Node processing
+The node responsibility can be described as: 
+> Graph node gets a fragment, processes it and responds with Transition. 
+So a node is the function:
+```
+F -> (F', T)
+```
+where `F` is Fragment, `F'` is modified Fragment, `T` is Transition.
+
+The node definition is abstract. It allows to define simple processing nodes but also more complex 
+structures such as a list of subgraphs. Furthermore, such a definition inspires to provide custom 
+node implementations.
+
+The `node` configuration is simple:
+```hocon
+graph {
+  node {
+    factory = action
     config {
-      endpointOptions {
-        path = /product/id
-        domain = webapi
-        port = 8080
-        allowedRequestHeaders = ["Content-Type"]
-      }
+      action = book-rest-api
     }
   }
-  # simple action with fallback
-  product-fallback {
-    factory = "inline-body"
+}
+
+
+```
+The `factory` parameter specifies a node factory name, `config` contains all options passed to 
+the factory. 
+
+Knot.x provides two node implementations:
+**Action node** that represents simple steps in a graph such as integration with a data source
+**SubTasks node** that is a list of unnamed tasks (sub-tasks) that are evaluated in parallel
+
+##### Action node
+Action node declares [Action](#actions) to execute by action name. Its configuration is simple:
+```hocon
+graph {
+  node {
+    factory = action
     config {
-      body = <div>Product not available at the moment</div>
+      action = book-rest-api
     }
   }
 }
 ```
+The above example specifies the action node that delegates processing to the `book-rest-api` action.
 
-## Task
-Task defines a directed Actions graph. Task definition (JSON) is transformed to graph of nodes: 
-- [ActionNode](https://github.com/Knotx/knotx-fragments/blob/master/handler/engine/src/main/java/io/knotx/fragments/engine/graph/ActionNode.java)
-- [CompositeNode](https://github.com/Knotx/knotx-fragments/blob/master/handler/engine/src/main/java/io/knotx/fragments/engine/graph/CompositeNode.java). 
-
-The example `pdp` task defines two action nodes. 
-
-A composite node is defined as a subgraph:
-```
-book-and-author {
-  actions = [
-    { action = book },
-    { action = author }
-  ]
-  onTransition {
-    _error {
-      action = book-and-author-fallback
+Knot.x allows simplifying action nodes declaration:
```

Comment:
If we intend to keep the legacy as a shorthand, maybe it's better to introduce them first, and then explain what's lying underneath to make things easier for begginers? (the same for `tasks` above)

Reply:
@Mateusz512 , @Voycawojka can you provide an example in a separate PR?

Reply:
[done](https://github.com/Knotx/knotx-fragments/issues/53)